lib/workers/repository/update/branch/bump-versions.ts

```diff
@@ -0,0 +1,215 @@
+  }
+
+  // skip if no packageFiles or artifacts have been updated
+  if (!config.updatedPackageFiles?.length && !config.updatedArtifacts?.length) {
+    return;
+  }
+
+  const allFiles = await scm.getFileList();
+  const fileList = getFilteredFileList(config, allFiles);
+
+  const packageFileChanges = fileChangeListToMap(config.updatedPackageFiles);
+  const artifactFileChanges = fileChangeListToMap(config.updatedArtifacts);
+
+  for (const bumpVersionConfig of bumpVersions) {
+    await bumpVersion(
+      bumpVersionConfig,
+      config,
+      fileList,
+      packageFileChanges.additions,
+      artifactFileChanges.additions,
+    );
+  }
+
+  // update the config with the new files
+  config.updatedPackageFiles = [
+    ...Object.values(packageFileChanges.additions),
+    ...Object.values(packageFileChanges.deletions),
+  ];
+  config.updatedArtifacts = [
+    ...Object.values(artifactFileChanges.additions),
+    ...Object.values(artifactFileChanges.deletions),
+  ];
+}
+
+async function bumpVersion(
+  config: BumpVersionConfig,
+  branchConfig: BranchConfig,
+  fileList: string[],
+  packageFiles: Record<string, FileAddition>,
+  artifactFiles: Record<string, FileAddition>,
+): Promise<void> {
+  const files: string[] = [];
+  try {
+    files.push(...getMatchedFiles(config.fileMatch, branchConfig, fileList));
+  } catch (e) {
+    addArtifactError(
+      branchConfig,
+      `Failed to calculate matched files for bumpVersions: ${e.message}`,
+    );
+  }
+
+  // prepare the matchStrings
+  const matchStringsRegexes: RegExp[] = [];
+  for (const matchString of config.matchStrings) {
+    try {
+      const templated = compile(matchString, branchConfig);
+      matchStringsRegexes.push(regEx(templated));
+    } catch (e) {
+      addArtifactError(
+        branchConfig,
+        `Failed to compile matchString for bumpVersions: ${e.message}`,
+        matchString,
+      );
+    }
+  }
+
+  for (const file of files) {
+    // getting the already modified file contents or read the file directly
+    let fileContents: string | null | undefined =
+      packageFiles[file]?.contents?.toString();
+    fileContents ??= artifactFiles[file]?.contents?.toString();
+    fileContents ??= await readLocalFile(file, 'utf8');
+    if (!fileContents) {
+      logger.trace({ file }, `bumpVersions: Could not read file`);
+      continue;
+    }
+
+    for (const matchStringRegex of matchStringsRegexes) {
+      // extracting the version from the file
+      const regexResult = matchStringRegex.exec(fileContents);
+      if (!regexResult) {
+        continue;
+      }
+      const version = regexResult.groups?.version;
+      if (!version) {
+        logger.trace({ file }, `bumpVersions: No version found`);
+        continue;
+      }
+
+      // getting new version
+      let newVersion: string | null = null;
+      try {
+        const bumpType = compile(config.bumpType, branchConfig);
+        newVersion = inc(version, bumpType as ReleaseType);
```

Comment:
we should document that only semver is supported

Reply:
This document via [here](https://github.com/renovatebot/renovate/pull/34023/files#diff-be0a39f342380493f17c380bbd694f1f05d18bdd8d7cecf174369084952102deR459)